src/app/shared/courses/courses.service.ts

```diff
@@ -22,32 +41,37 @@ export class CoursesService {
         return this.orderByName.transform(this.getList(page, displayLimit));
     }
 
-    createCourse(title: string, duration: number, description: string, topRated: boolean): Course {
+    createCourse(name: string, length: number, description: string, isTopRated: boolean): Course {
         const course = {
             id: this.getCoursesLength() + 1,
-            title,
-            dateCreation: new Date(),
-            duration,
+            name,
+            date: new Date(),
+            length,
             description,
-            topRated
+            authors: null,
+            isTopRated
         };
-        this.courses.push(course);
+        this.http.post('http://localhost:3004/courses/', course, httpOptions);
         return course;
     }
 
-    update(id: number, title: string, duration: number, description: string, topRated: boolean): Course {
+    update(id: number, name: string, length: number, description: string, isTopRated: boolean): Course {
         const course: Course = this.getItemById(id);
-        course.title = title;
-        course.duration = duration;
+        course.name = name;
+        course.length = length;
         course.description = description;
-        course.topRated = topRated;
+        course.isTopRated = isTopRated;
         return course;
     }
 
     removeCourse(id: number): Course[] {
         return this.courses = this.courses.filter(course => course.id !== id);
     }
 
+    removeServerCourse(id: number): void {
+        this.http.delete(`http://localhost:3004/courses/${id}`);
+    }
```

Comment:
consider using only single api:
```typescript
removeCourse(id: number): Observable<any> {
  return this.http.delete(`http://localhost:3004/courses/${ id }`);
}
```
Here is no need to use extra state properties. Now state is held on the server.